dotnet/src/SemanticKernel.Core/Functions/SemanticFunction.cs

```diff
@@ -217,15 +211,36 @@ private void AddDefaultValues(ContextVariables variables)
         try
         {
             string renderedPrompt = await this._promptTemplate.RenderAsync(context, cancellationToken).ConfigureAwait(false);
-            IReadOnlyList<ITextResult> completionResults = await client.GetCompletionsAsync(renderedPrompt, requestSettings, cancellationToken).ConfigureAwait(false);
-            string completion = await GetCompletionsResultContentAsync(completionResults, cancellationToken).ConfigureAwait(false);
+            var modelResults = new List<ModelResult>();
 
-            // Update the result with the completion
-            context.Variables.Update(completion);
-
-            result = new FunctionResult(this.Name, this.PluginName, context, completion);
-
-            var modelResults = completionResults.Select(c => c.ModelResult).ToArray();
+            if (requestSettings?.Streaming == true &&
```

Comment:
I would suggest to move streaming and non-streaming logic to separate private methods to keep method `RunPromptAsync` as simple as possible.